components/abstractions/gradle/dependencies.gradle

```diff
@@ -10,7 +10,7 @@ dependencies {
     // This dependency is used internally, and not exposed to consumers on their own compile classpath.
     implementation 'com.google.guava:guava:32.1.2-jre'
     implementation 'org.javatuples:javatuples:1.2'
-    implementation 'javax.annotation:javax.annotation-api:1.3.2'
+    implementation 'jakarta.annotation:jakarta.annotation-api:2.1.1'
```

Comment:
we might want to switch this one to API instead so consumers get the dependency automatically for the generated code. What do you think @andreaTP ?

Reply:
This would also mean that we don't have to specify the dependency for the other components since the abstractions would make it available for them.

Reply:
@baywet I'm not sure I'm following here, at the moment the dependency is on API afaiu, or am I missing something?

Reply:
correct, I know we've already had dependency management issues in the past. This is why I'm careful about making it an API dependency and would like @andreaTP point of view.
The alternative being to include the dependency in the kiota info command. It be a first since we don't require people to include anything else besides abstractions.

Reply:
@anreaTp I guess we were writing at the same time. Because we depend on that annotation for the generated code, people would need to add it to there project dependency for things to compile if we leave that as an implementation dependency.

Reply:
@baywet I'm about to log off for today, I think that what you are saying is not accurate.
At the moment we do ask the user to provide the findbugs dependency(at least for compile time).

I'm completely in favor of leaving this to the user, and, possibly, adding it to `kiota info` as you proposed.
Compile time dependency should not spill to the user runtime and the user should be in control of it.

Reply:
The different being [nullable/nonnull are runtime annotations](https://github.com/jakartaee/common-annotations-api/blob/0e6318ac2716c463b33cceff191c9639c73e80ca/api/src/main/java/jakarta/annotation/Nullable.java#L41) and findbugs [is a class annotation](https://github.com/spotbugs/spotbugs/blob/2fdbb01514683c32c07aeb841879269dddccbafc/spotbugs-annotations/src/main/java/edu/umd/cs/findbugs/annotations/SuppressFBWarnings.java#L33) so it gets erased earlier in the lifecycle.

We're getting rid of the suppress fb warnings annotations to trim down our dependencies (or the ones we're pushing on the consumer). So this won't be a problem for long.

Let's play the worst case scenarios for both options:
- implementation, people don't add it to their dependencies, I'm guessing they'd get an error at compile time saying they are missing a dependency?
- api, people have a different version brought by another dependency, now they are stuck with a conflict that's hard to resolve

Am I understanding this correctly?

Reply:
I completely switch api/implementation with the respective worst case 🙂 .
Let's talk about the concrete example:

`jakarta.annotation:jakarta.annotation-api:2.1.1`

This is shipping the API but not the implementation, people will be asked to provide it on the classpath.
I think that keeping this is perfectly fine.

Reply:
alright @ramsessanchez can you add that dependency to the appsettings in your kiota PR please?
Thanks andrea for the help here

Reply:
yep, leave implementation as is, add an entry in the appsettings.json file in kiota. Thanks